src/snowcli/cli/snowpark_shared.py

```diff
@@ -81,31 +77,35 @@ def snowpark_package(
             )
             if do_download:
                 log.info("Installing non-Anaconda packages...")
-                should_pack, second_chance_results = utils.install_packages(
+                should_continue, second_chance_results = utils.install_packages(
                     "requirements.other.txt",
                     check_anaconda_for_pypi_deps,
                     package_native_libraries,
                 )
-                if should_pack:
+                if should_continue:
                     # add the Anaconda packages discovered as dependencies
-                    if second_chance_results is not None:
+                    if second_chance_results:
                         split_requirements.snowflake = (
                             split_requirements.snowflake
                             + second_chance_results.snowflake
                         )
 
         # write requirements.snowflake.txt file
         if split_requirements.snowflake:
-            log.info("Writing requirements.snowflake.txt file...")
-            with open(
+            _write_requirements_file(
                 "requirements.snowflake.txt",
-                "w",
-                encoding="utf-8",
-            ) as f:
-                for package in utils.deduplicate_and_sort_reqs(
-                    split_requirements.snowflake
-                ):
-                    f.write(package.line + "\n")
+                utils.deduplicate_and_sort_reqs(split_requirements.snowflake),
+            )
 
     zip_dir(source=source, dest_zip=artefact_file)
-    log.info("Deployment package now ready: app.zip")
+
+    if os.path.exists(".packages"):
```

Comment:
I think we should simply override existing packages, this is like a build command that replaces previous build artifacts. WDYT?

Reply:
But this part is not overriding, but adding to app.zip